spec/requests/admin/supplier_users_spec.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Admin::SupplierUsers", type: :request do
+  let(:lead_provider) { create(:lead_provider) }
+  let(:delivery_partner) { create(:delivery_partner) }
+  let(:full_name) { Faker::Name.name }
+  let(:email) { Faker::Internet.email }
+
+  before do
+    admin_user = create(:user, :admin)
+    sign_in admin_user
+  end
+
+  describe "GET /admin/suppliers/users" do
+    it "renders the index template" do
+      get "/admin/suppliers/users"
+
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "GET /admin/suppliers/users/new" do
+    it "renders the new template" do
+      get "/admin/suppliers/users/new"
+
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "POST /admin/suppliers/new" do
+    it "redirects to the supplier type page" do
+      when_i_choose_a_supplier(lead_provider)
+
+      expect(response).to redirect_to("/admin/suppliers/users/new/user-details")
+    end
+
+    it "Sets the correct lead provider" do
+      when_i_choose_a_supplier(lead_provider)
+
+      # Then
```

Comment:
Gherkin language ;) Kudos for that - the comment can be removed though - the tests are self-explanatory